java-agent/instrumentation/servlet/src/main/java/io/opentelemetry/auto/instrumentation/servlet/filter/FilterInstrumentation.java

```diff
@@ -59,30 +57,32 @@ public boolean defaultEnabled() {
   public static class FilterAdvice {
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static AgentScope start(@Advice.This final Filter filter) {
-      if (activeSpan() == null) {
+    public static SpanScopePair start(@Advice.This final Filter filter) {
+      if (TRACER.getCurrentSpan() == null || !TRACER.getCurrentSpan().getContext().isValid()) {
```

Comment:
```suggestion
      if (!TRACER.getCurrentSpan().getContext().isValid()) {
```
TRACER.getCurrentSpan() never returns `null`